lib/cylc/task_events_mgr.py

```diff
@@ -757,6 +776,7 @@ def _process_message_submitted(self, itask, event_time):
             if itask.state.reset_state(TASK_STATUS_SUBMITTED):
                 self.setup_event_handlers(
                     itask, TASK_OUTPUT_SUBMITTED, 'job submitted')
+                self.job_pool.set_job_state(job_d, TASK_STATUS_SUBMITTED)
             self._reset_job_timers(itask)
```

Comment:
There are a few calls (e.g. `self.job_pool.set_job_time`, `self.job_pool.set_job_attr`) that are receiving string values, that may change later... 

Instead we could either get an object from the pool, or use constants in the method call I think... though not a blocker IMO.

Reply:
Think is; these are spread over different methods (often a time and state in one), and the format may need changing when being set or a test done to make sure the object exists, i.e.:
```
    def set_job_time(self, job_d, event_key, time_str=None):
        """Set an event time in job pool object.

        Set values of both event_key + "_time" and event_key + "_time_string".
        """
        try:
            setattr(self.pool[job_d], event_key + '_time', time_str)
        except KeyError:
            pass
```
As might be possible to reset a task state or something when a job script hasn't been written yet.